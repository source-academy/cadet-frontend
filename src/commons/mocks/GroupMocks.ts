import { GradingSummary } from '../../features/dashboard/DashboardTypes';

export const mockGradingSummary: GradingSummary = [
  {
    leaderName: 'John',
    groupName: 'Mock Group 1',
    ungradedMissions: 123,
    submittedMissions: 200,
    ungradedSidequests: 100,
    submittedSidequests: 117
  },
  {
    leaderName: 'Molly',
    groupName: 'Mock Group 2',
    ungradedMissions: 1232,
    submittedMissions: 205430,
    ungradedSidequests: 345,
    submittedSidequests: 11547
  },
  {
    leaderName: 'Lenny',
    groupName: 'Mock Group 3',
    ungradedMissions: 1532,
    submittedMissions: 22200,
    ungradedSidequests: 134500,
    submittedSidequests: 6777
  }
];
