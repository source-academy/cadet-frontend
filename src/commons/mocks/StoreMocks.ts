import { Store } from 'redux';
import mockStore from 'redux-mock-store';

import {
  defaultAcademy,
  defaultApplication,
  defaultDashboard,
  defaultPlayground,
  defaultSession,
  defaultWorkspaceManager,
  OverallState
} from '../application/ApplicationTypes';

export function mockInitialStore<P>(): Store<OverallState> {
  const createStore = (mockStore as any)();
  const state: OverallState = {
    academy: defaultAcademy,
    application: defaultApplication,
    dashboard: defaultDashboard,
    playground: defaultPlayground,
    workspaces: defaultWorkspaceManager,
    session: defaultSession
  };
  return createStore(state);
}
