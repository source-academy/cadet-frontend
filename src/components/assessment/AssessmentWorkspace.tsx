import {
  Button,
  ButtonGroup,
  Card,
  Classes,
  Dialog,
  Intent,
  NonIdealState,
  Spinner
} from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import * as classNames from 'classnames';
import * as React from 'react';
import ChatApp from '../../containers/ChatContainer';
import { InterpreterOutput, IWorkspaceState } from '../../reducers/states';
import { USE_CHATKIT } from '../../utils/constants';
import { beforeNow } from '../../utils/dateHelpers';
import { history } from '../../utils/history';
import { assessmentCategoryLink } from '../../utils/paramParseHelpers';
import { controlButton } from '../commons';
import Markdown from '../commons/Markdown';
import Workspace, { WorkspaceProps } from '../workspace';
import { ControlBarProps } from '../workspace/ControlBar';
import { SideContentProps } from '../workspace/side-content';
import Autograder from '../workspace/side-content/Autograder';
import ToneMatrix from '../workspace/side-content/ToneMatrix';
import {
  AutogradingResult,
  IAssessment,
  IMCQQuestion,
  IProgrammingQuestion,
  IQuestion,
  ITestcase,
  Library,
  QuestionTypes
} from './assessmentShape';
import GradingResult from './GradingResult';

export type AssessmentWorkspaceProps = DispatchProps & OwnProps & StateProps;

export type StateProps = {
  activeTab: number;
  assessment?: IAssessment;
  autogradingResults: AutogradingResult[];
  editorPrepend: string;
  editorValue: string | null;
  editorPostpend: string;
  editorTestcases: ITestcase[];
  editorHeight?: number;
  editorWidth: string;
  breakpoints: string[];
  highlightedLines: number[][];
  hasUnsavedChanges: boolean;
  isRunning: boolean;
  isDebugging: boolean;
  enableDebugging: boolean;
  output: InterpreterOutput[];
  replValue: string;
  sideContentHeight?: number;
  storedAssessmentId?: number;
  storedQuestionId?: number;
};

export type OwnProps = {
  assessmentId: number;
  questionId: number;
  notAttempted: boolean;
  closeDate: string;
};

export type DispatchProps = {
  handleAssessmentFetch: (assessmentId: number) => void;
  handleBrowseHistoryDown: () => void;
  handleBrowseHistoryUp: () => void;
  handleChangeActiveTab: (activeTab: number) => void;
  handleChapterSelect: (chapter: any, changeEvent: any) => void;
  handleClearContext: (library: Library) => void;
  handleEditorEval: () => void;
  handleEditorValueChange: (val: string) => void;
  handleEditorHeightChange: (height: number) => void;
  handleEditorWidthChange: (widthChange: number) => void;
  handleEditorUpdateBreakpoints: (breakpoints: string[]) => void;
  handleInterruptEval: () => void;
  handleReplEval: () => void;
  handleReplOutputClear: () => void;
  handleReplValueChange: (newValue: string) => void;
  handleResetWorkspace: (options: Partial<IWorkspaceState>) => void;
  handleSave: (id: number, answer: number | string) => void;
  handleSideContentHeightChange: (heightChange: number) => void;
  handleTestcaseEval: (testcaseId: number) => void;
  handleUpdateCurrentAssessmentId: (assessmentId: number, questionId: number) => void;
  handleUpdateHasUnsavedChanges: (hasUnsavedChanges: boolean) => void;
  handleDebuggerPause: () => void;
  handleDebuggerResume: () => void;
  handleDebuggerReset: () => void;
};

class AssessmentWorkspace extends React.Component<
  AssessmentWorkspaceProps,
  { showOverlay: boolean; showResetTemplateOverlay: boolean }
> {
  public constructor(props: AssessmentWorkspaceProps) {
    super(props);
    this.state = {
      showOverlay: false,
      showResetTemplateOverlay: false
    };
    this.props.handleEditorValueChange('');
  }

  /**
   * After mounting (either an older copy of the assessment
   * or a loading screen), try to fetch a newer assessment,
   * and show the briefing.
   */
  public componentDidMount() {
    this.props.handleAssessmentFetch(this.props.assessmentId);
    if (this.props.questionId === 0 && this.props.notAttempted) {
      this.setState({ showOverlay: true });
    }
    if (!this.props.assessment) {
      return;
    }

    let questionId = this.props.questionId;
    if (this.props.questionId >= this.props.assessment.questions.length) {
      questionId = this.props.assessment.questions.length - 1;
    }

    const question: IQuestion = this.props.assessment.questions[questionId];

    let answer = '';
    if (question.type === QuestionTypes.programming) {
      if (question.answer) {
        answer = (question as IProgrammingQuestion).answer as string;
      } else {
        answer = (question as IProgrammingQuestion).solutionTemplate;
      }
    }

    this.props.handleEditorValueChange(answer);
  }

  /**
   * Once there is an update (due to the assessment being fetched), check
   * if a workspace reset is needed.
   */
  public componentDidUpdate() {
    this.checkWorkspaceReset(this.props);
  }

  public render() {
    if (this.props.assessment === undefined || this.props.assessment.questions.length === 0) {
      return (
        <NonIdealState
          className={classNames('WorkspaceParent', Classes.DARK)}
          description="Getting mission ready..."
          icon={<Spinner size={Spinner.SIZE_LARGE} />}
        />
      );
    }
    const overlay = (
      <Dialog className="assessment-briefing" isOpen={this.state.showOverlay}>
        <Card>
          <Markdown content={this.props.assessment.longSummary} />
          <Button
            className="assessment-briefing-button"
            // tslint:disable-next-line jsx-no-lambda
            onClick={() => this.setState({ showOverlay: false })}
            text="Continue"
          />
        </Card>
      </Dialog>
    );

    const resetTemplateOverlay = (
      <Dialog
        className="assessment-reset"
        icon={IconNames.ERROR}
        isCloseButtonShown={false}
        isOpen={this.state.showResetTemplateOverlay}
        title="Confirmation: Reset editor?"
      >
        <div className={Classes.DIALOG_BODY}>
          <Markdown content="Are you sure you want to reset the template?" />
          <Markdown content="*Note this will not affect the saved copy of your code, unless you save over it.*" />
        </div>
        <div className={Classes.DIALOG_FOOTER}>
          <ButtonGroup>
            {controlButton(
              'Cancel',
              null,
              () => this.setState({ showResetTemplateOverlay: false }),
              {
                minimal: false
              }
            )}
            {controlButton(
              'Confirm',
              null,
              () => {
                this.setState({ showResetTemplateOverlay: false });
                this.props.handleEditorValueChange(
                  (this.props.assessment!.questions[questionId] as IProgrammingQuestion)
                    .solutionTemplate
                );
                this.props.handleUpdateHasUnsavedChanges(true);
              },
              { minimal: false, intent: Intent.DANGER }
            )}
          </ButtonGroup>
        </div>
      </Dialog>
    );
    /* If questionId is out of bounds, set it to the max. */
    const questionId =
      this.props.questionId >= this.props.assessment.questions.length
        ? this.props.assessment.questions.length - 1
        : this.props.questionId;
    const question: IQuestion = this.props.assessment.questions[questionId];
    const workspaceProps: WorkspaceProps = {
      controlBarProps: this.controlBarProps(this.props, questionId),
      editorProps:
        question.type === QuestionTypes.programming
          ? {
              editorSessionId: '',
              editorPrepend: this.props.editorPrepend,
              editorPrependLines:
                this.props.editorPrepend.length === 0
                  ? 0
                  : this.props.editorPrepend.split('\n').length,
              editorValue: this.props.editorValue!,
              handleEditorEval: this.props.handleEditorEval,
              handleEditorValueChange: this.props.handleEditorValueChange,
              handleUpdateHasUnsavedChanges: this.props.handleUpdateHasUnsavedChanges,
              breakpoints: this.props.breakpoints,
              highlightedLines: this.props.highlightedLines,
              handleEditorUpdateBreakpoints: this.props.handleEditorUpdateBreakpoints,
              isEditorAutorun: false
            }
          : undefined,
      editorHeight: this.props.editorHeight,
      editorWidth: this.props.editorWidth,
      handleEditorHeightChange: this.props.handleEditorHeightChange,
      handleEditorWidthChange: this.props.handleEditorWidthChange,
      handleSideContentHeightChange: this.props.handleSideContentHeightChange,
      hasUnsavedChanges: this.props.hasUnsavedChanges,
      mcqProps: {
        mcq: question as IMCQQuestion,
        handleMCQSubmit: (option: number) =>
          this.props.handleSave(this.props.assessment!.questions[questionId].id, option)
      },
      sideContentHeight: this.props.sideContentHeight,
      sideContentProps: this.sideContentProps(this.props, questionId),
      replProps: {
        handleBrowseHistoryDown: this.props.handleBrowseHistoryDown,
        handleBrowseHistoryUp: this.props.handleBrowseHistoryUp,
        handleReplEval: this.props.handleReplEval,
        handleReplValueChange: this.props.handleReplValueChange,
        output: this.props.output,
        replValue: this.props.replValue
      }
    };
    return (
      <div className={classNames('WorkspaceParent', Classes.DARK)}>
        {overlay}
        {resetTemplateOverlay}
        <Workspace {...workspaceProps} />
      </div>
    );
  }

  /**
   * Checks if there is a need to reset the workspace, then executes
   * a dispatch (in the props) if needed.
   */
  private checkWorkspaceReset(props: AssessmentWorkspaceProps) {
    /* Don't reset workspace if assessment not fetched yet. */
    if (this.props.assessment === undefined) {
      return;
    }

    /* Reset assessment if it has changed.*/
    const assessmentId = this.props.assessmentId;
    const questionId = this.props.questionId;

    if (
      this.props.storedAssessmentId === assessmentId &&
      this.props.storedQuestionId === questionId
    ) {
      return;
    }

    const question = this.props.assessment.questions[questionId];

    let autogradingResults: AutogradingResult[] = [];
    let editorValue: string = '';
    let editorPrepend: string = '';
    let editorPostpend: string = '';
    let editorTestcases: ITestcase[] = [];

    if (question.type === QuestionTypes.programming) {
      const questionData = question as IProgrammingQuestion;
      autogradingResults = questionData.autogradingResults;
      editorPrepend = questionData.prepend;
      editorPostpend = questionData.postpend;
      editorTestcases = questionData.testcases;

      editorValue = questionData.answer as string;
      if (!editorValue) {
        editorValue = questionData.solutionTemplate!;
      }
    }

    this.props.handleEditorUpdateBreakpoints([]);
    this.props.handleUpdateCurrentAssessmentId(assessmentId, questionId);
    this.props.handleResetWorkspace({
      autogradingResults,
      editorPrepend,
      editorValue,
      editorPostpend,
      editorTestcases
    });
    this.props.handleClearContext(question.library);
    this.props.handleUpdateHasUnsavedChanges(false);
    if (editorValue) {
      this.props.handleEditorValueChange(editorValue);
    }
  }

  /** Pre-condition: IAssessment has been loaded */
  private sideContentProps: (p: AssessmentWorkspaceProps, q: number) => SideContentProps = (
    props: AssessmentWorkspaceProps,
    questionId: number
  ) => {
    const tabs = [
      {
        label: `Task ${questionId + 1}`,
        icon: IconNames.NINJA,
        body: <Markdown content={props.assessment!.questions[questionId].content} />
      },
      {
        label: `${props.assessment!.category} Briefing`,
        icon: IconNames.BRIEFCASE,
        body: <Markdown content={props.assessment!.longSummary} />
      },
      {
        label: `${props.assessment!.category} Autograder`,
        icon: IconNames.AIRPLANE,
        body: (
          <Autograder
            testcases={props.editorTestcases}
            autogradingResults={props.autogradingResults}
            handleTestcaseEval={this.props.handleTestcaseEval}
          />
        )
      }
    ];
    const isGraded = props.assessment!.questions[questionId].grader !== null;
    if (isGraded) {
      tabs.push(
        {
          label: `Grading`,
          icon: IconNames.TICK,
          body: (
            <GradingResult
              graderName={props.assessment!.questions[questionId].grader.name}
              gradedAt={props.assessment!.questions[questionId].gradedAt}
              xp={props.assessment!.questions[questionId].xp}
              grade={props.assessment!.questions[questionId].grade}
              maxGrade={props.assessment!.questions[questionId].maxGrade}
              maxXp={props.assessment!.questions[questionId].maxXp}
            />
          )
        },
        {
          label: `Comments`,
          icon: IconNames.CHAT,
          body: USE_CHATKIT ? (
            <ChatApp roomId={props.assessment!.questions[questionId].comment} />
          ) : (
            <span>Chatkit disabled.</span>
          )
        }
      );
    }

    const functionsAttached = props.assessment!.questions[questionId].library.external.symbols;
    if (functionsAttached.includes('get_matrix')) {
      tabs.push({
        label: `Tone Matrix`,
        icon: IconNames.GRID_VIEW,
        body: <ToneMatrix />
      });
    }
    return {
      activeTab: props.activeTab,
      handleChangeActiveTab: props.handleChangeActiveTab,
      tabs
    };
  };

  /** Pre-condition: IAssessment has been loaded */
  private controlBarProps: (p: AssessmentWorkspaceProps, q: number) => ControlBarProps = (
    props: AssessmentWorkspaceProps,
    questionId: number
  ) => {
    const listingPath = `/academy/${assessmentCategoryLink(this.props.assessment!.category)}`;
    const assessmentWorkspacePath = listingPath + `/${this.props.assessment!.id.toString()}`;
    return {
      handleChapterSelect: this.props.handleChapterSelect,
      handleEditorEval: this.props.handleEditorEval,
      handleInterruptEval: this.props.handleInterruptEval,
      handleReplEval: this.props.handleReplEval,
      handleReplOutputClear: this.props.handleReplOutputClear,
      handleReplValueChange: this.props.handleReplValueChange,
      handleDebuggerPause: this.props.handleDebuggerPause,
      handleDebuggerResume: this.props.handleDebuggerResume,
      handleDebuggerReset: this.props.handleDebuggerReset,
      hasChapterSelect: false,
      hasCollabEditing: false,
      hasEditorAutorunButton: false,
      hasSaveButton:
        !beforeNow(this.props.closeDate) &&
        this.props.assessment!.questions[questionId].type !== QuestionTypes.mcq,
      hasShareButton: false,
      isRunning: this.props.isRunning,
      isDebugging: this.props.isDebugging,
      enableDebugging: this.props.enableDebugging,
      onClickNext: () => history.push(assessmentWorkspacePath + `/${(questionId + 1).toString()}`),
      onClickPrevious: () =>
        history.push(assessmentWorkspacePath + `/${(questionId - 1).toString()}`),
      onClickReturn: () => history.push(listingPath),
      onClickSave: () =>
        this.props.handleSave(
          this.props.assessment!.questions[questionId].id,
          this.props.editorValue!
        ),
      onClickResetTemplate: () => {
        this.setState({ showResetTemplateOverlay: true });
      },
      questionProgress: [questionId + 1, this.props.assessment!.questions.length],
      sourceChapter: this.props.assessment!.questions[questionId].library.chapter
    };
  };
}

export default AssessmentWorkspace;
