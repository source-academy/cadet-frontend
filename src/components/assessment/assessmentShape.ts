/*
 * Used to display information regarding an assessment in the UI.
 *
 * @property closeAt an ISO 8601 compliant date string specifiying when
 *   the assessment closes
 * @property openAt an ISO 8601 compliant date string specifiying when
 *   the assessment opens
 */
export interface IAssessmentOverview {
  category: AssessmentCategory;
  closeAt: string;
  coverImage: string;
  grade: number;
  id: number;
  maxGrade: number;
  maxXp: number;
  openAt: string;
  title: string;
  shortSummary: string;
  status: AssessmentStatus;
  story: string | null;
  xp: number;
  gradingStatus: GradingStatus;
}

export enum AssessmentStatuses {
  not_attempted = 'not_attempted',
  attempting = 'attempting',
  attempted = 'attempted',
  submitted = 'submitted'
}

export type AssessmentStatus = keyof typeof AssessmentStatuses;

export enum GradingStatuses {
  none = 'none',
  grading = 'grading',
  graded = 'graded'
}

export type GradingStatus = keyof typeof GradingStatuses;

/*
 * Used when an assessment is being actively attempted/graded.
 */
export interface IAssessment {
  category: AssessmentCategory;
  id: number;
  longSummary: string;
  missionPDF: string;
  questions: IQuestion[];
  title: string;
}

/* The different kinds of Assessments available */
export enum AssessmentCategories {
  Contest = 'Contest',
  Mission = 'Mission',
  Path = 'Path',
  Sidequest = 'Sidequest'
}

export type AssessmentCategory = keyof typeof AssessmentCategories;

export interface IProgrammingQuestion extends IQuestion {
  answer: string | null;
  solutionTemplate: string;
  type: 'programming';
}

export interface IMCQQuestion extends IQuestion {
  solution: number | null;
  answer: number | null;
  choices: MCQChoice[];
  type: 'mcq';
}

export interface IQuestion {
  answer: string | number | null;
  comment: string | null;
  content: string;
  id: number;
  library: Library;
  type: QuestionType;
  grader: {
    name: string;
    id: number;
  };
  gradedAt: string;
  xp: number;
  grade: number;
  maxGrade: number;
  maxXp: number;
}

export type MCQChoice = {
  content: string;
  hint: string | null;
};

/* The two kinds of Questions available */
export enum QuestionTypes {
  programming = 'programming',
  mcq = 'mcq'
}
export type QuestionType = keyof typeof QuestionTypes;

/** Constants for external library names */
export enum ExternalLibraryNames {
  NONE = 'NONE',
  TWO_DIM_RUNES = 'TWO_DIM_RUNES',
  THREE_DIM_RUNES = 'THREE_DIM_RUNES',
  CURVES = 'CURVES',
  SOUND = 'SOUND',
  STREAMS = 'STREAMS'
}

export type ExternalLibraryName = keyof typeof ExternalLibraryNames;

type ExternalLibrary = {
  name: ExternalLibraryName;
  symbols: string[];
};

export type Library = {
  chapter: number;
  external: ExternalLibrary;
  globals: Array<[string, any]>;
};
