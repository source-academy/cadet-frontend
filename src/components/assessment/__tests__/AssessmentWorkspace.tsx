import { shallow } from 'enzyme';
import * as React from 'react';

import { mockAssessments } from '../../../mocks/assessmentAPI';
import { Library } from '../assessmentShape';
import AssessmentWorkspace, { AssessmentWorkspaceProps } from '../AssessmentWorkspace';

const defaultProps: AssessmentWorkspaceProps = {
  activeTab: 0,
  assessmentId: 0,
  autogradingResults: [],
  notAttempted: true,
  closeDate: '2048-06-18T05:24:26.026Z',
  editorPrepend: '',
  editorValue: null,
  editorPostpend: '',
  editorTestcases: [],
  editorWidth: '50%',
  breakpoints: [],
  highlightedLines: [],
  hasUnsavedChanges: false,
  handleAssessmentFetch: (assessmentId: number) => {},
  handleBrowseHistoryDown: () => {},
  handleBrowseHistoryUp: () => {},
  handleChangeActiveTab: (activeTab: number) => {},
  handleChapterSelect: (chapter: any, changeEvent: any) => {},
  handleClearContext: (library: Library) => {},
  handleEditorEval: () => {},
  handleEditorValueChange: (val: string) => {},
  handleEditorHeightChange: (height: number) => {},
  handleEditorWidthChange: (widthChange: number) => {},
  handleEditorUpdateBreakpoints: (breakpoints: string[]) => {},
  handleInterruptEval: () => {},
  handleReplEval: () => {},
  handleReplOutputClear: () => {},
  handleReplValueChange: (newValue: string) => {},
  handleResetWorkspace: () => {},
  handleSave: (id: number, answer: string | number) => {},
  handleSideContentHeightChange: (heightChange: number) => {},
  handleTestcaseEval: (testcaseId: number) => {},
  handleUpdateHasUnsavedChanges: (hasUnsavedChanges: boolean) => {},
  handleUpdateCurrentAssessmentId: (a: number, q: number) => {},
  handleDebuggerPause: () => {},
  handleDebuggerResume: () => {},
  handleDebuggerReset: () => {},
  isRunning: false,
  isDebugging: false,
  enableDebugging: false,
  output: [],
  questionId: 0,
  replValue: ''
};

const mockUndefinedAssessmentWorkspaceProps: AssessmentWorkspaceProps = {
  ...defaultProps
};

const mockProgrammingAssessmentWorkspaceProps: AssessmentWorkspaceProps = {
  ...defaultProps,
  assessment: mockAssessments[0],
  assessmentId: 0,
  questionId: 0
};

const mockClosedProgrammingAssessmentWorkspaceProps: AssessmentWorkspaceProps = {
  ...mockProgrammingAssessmentWorkspaceProps,
  closeDate: '2008-06-18T05:24:26.026Z'
};

const mockMcqAssessmentWorkspaceProps: AssessmentWorkspaceProps = {
  ...defaultProps,
  assessment: mockAssessments[0],
  assessmentId: 0,
  questionId: 2
};

test('AssessmentWorkspace page "loading" content renders correctly', () => {
  const app = <AssessmentWorkspace {...mockUndefinedAssessmentWorkspaceProps} />;
  const tree = shallow(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('AssessmentWorkspace page with programming question renders correctly', () => {
  const app = <AssessmentWorkspace {...mockProgrammingAssessmentWorkspaceProps} />;
  const tree = shallow(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('AssessmentWorkspace page with overdue assessment renders correctly', () => {
  const app = <AssessmentWorkspace {...mockClosedProgrammingAssessmentWorkspaceProps} />;
  const tree = shallow(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('AssessmentWorkspace page with MCQ question renders correctly', () => {
  const app = <AssessmentWorkspace {...mockMcqAssessmentWorkspaceProps} />;
  const tree = shallow(app);
  expect(tree.debug()).toMatchSnapshot();
});
