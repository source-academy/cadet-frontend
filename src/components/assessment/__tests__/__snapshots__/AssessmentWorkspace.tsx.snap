// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentWorkspace page "loading" content renders correctly 1`] = `"<NonIdealState className=\\"WorkspaceParent pt-dark\\" description=\\"Getting mission ready...\\" visual={{...}} />"`;

exports[`AssessmentWorkspace page with MCQ question renders correctly 1`] = `
"<div className=\\"WorkspaceParent pt-dark\\">
  <Blueprint2.Dialog className=\\"assessment-briefing\\" isOpen={false} canOutsideClickClose={true}>
    <Blueprint2.Card elevation={0} interactive={false}>
      <Markdown content=\\"This is the mission briefing. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas viverra, sem scelerisque ultricies ullamcorper, sem nibh sollicitudin enim, at ultricies sem orci eget odio. Pellentesque varius et mauris quis vestibulum. Etiam in egestas dolor. Nunc consectetur, sapien sodales accumsan convallis, lectus mi tempus ipsum, vel ornare metus turpis sed justo. Vivamus at tellus sed ex convallis commodo at in lectus. Pellentesque pharetra pulvinar sapien pellentesque facilisis. Curabitur efficitur malesuada urna sed aliquam. Quisque massa metus, aliquam in sagittis non, cursus in sem. Morbi vel nunc at nunc pharetra lobortis. Aliquam feugiat ultricies ipsum vel sollicitudin. Vivamus nulla massa, hendrerit sit amet nibh quis, porttitor convallis nisi. \\" />
      <Blueprint2.Button className=\\"assessment-briefing-button\\" onClick={[Function: onClick]} text=\\"Continue\\" />
    </Blueprint2.Card>
  </Blueprint2.Dialog>
  <Workspace controlBarProps={{...}} editorProps={[undefined]} editorWidth=\\"50%\\" handleEditorWidthChange={[Function: handleEditorWidthChange]} handleSideContentHeightChange={[Function: handleSideContentHeightChange]} hasUnsavedChanges={false} mcqProps={{...}} sideContentHeight={[undefined]} sideContentProps={{...}} replProps={{...}} />
</div>"
`;

exports[`AssessmentWorkspace page with overdue assessment renders correctly 1`] = `
"<div className=\\"WorkspaceParent pt-dark\\">
  <Blueprint2.Dialog className=\\"assessment-briefing\\" isOpen={true} canOutsideClickClose={true}>
    <Blueprint2.Card elevation={0} interactive={false}>
      <Markdown content=\\"This is the mission briefing. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas viverra, sem scelerisque ultricies ullamcorper, sem nibh sollicitudin enim, at ultricies sem orci eget odio. Pellentesque varius et mauris quis vestibulum. Etiam in egestas dolor. Nunc consectetur, sapien sodales accumsan convallis, lectus mi tempus ipsum, vel ornare metus turpis sed justo. Vivamus at tellus sed ex convallis commodo at in lectus. Pellentesque pharetra pulvinar sapien pellentesque facilisis. Curabitur efficitur malesuada urna sed aliquam. Quisque massa metus, aliquam in sagittis non, cursus in sem. Morbi vel nunc at nunc pharetra lobortis. Aliquam feugiat ultricies ipsum vel sollicitudin. Vivamus nulla massa, hendrerit sit amet nibh quis, porttitor convallis nisi. \\" />
      <Blueprint2.Button className=\\"assessment-briefing-button\\" onClick={[Function: onClick]} text=\\"Continue\\" />
    </Blueprint2.Card>
  </Blueprint2.Dialog>
  <Workspace controlBarProps={{...}} editorProps={{...}} editorWidth=\\"50%\\" handleEditorWidthChange={[Function: handleEditorWidthChange]} handleSideContentHeightChange={[Function: handleSideContentHeightChange]} hasUnsavedChanges={false} mcqProps={{...}} sideContentHeight={[undefined]} sideContentProps={{...}} replProps={{...}} />
</div>"
`;

exports[`AssessmentWorkspace page with programming question renders correctly 1`] = `
"<div className=\\"WorkspaceParent pt-dark\\">
  <Blueprint2.Dialog className=\\"assessment-briefing\\" isOpen={true} canOutsideClickClose={true}>
    <Blueprint2.Card elevation={0} interactive={false}>
      <Markdown content=\\"This is the mission briefing. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas viverra, sem scelerisque ultricies ullamcorper, sem nibh sollicitudin enim, at ultricies sem orci eget odio. Pellentesque varius et mauris quis vestibulum. Etiam in egestas dolor. Nunc consectetur, sapien sodales accumsan convallis, lectus mi tempus ipsum, vel ornare metus turpis sed justo. Vivamus at tellus sed ex convallis commodo at in lectus. Pellentesque pharetra pulvinar sapien pellentesque facilisis. Curabitur efficitur malesuada urna sed aliquam. Quisque massa metus, aliquam in sagittis non, cursus in sem. Morbi vel nunc at nunc pharetra lobortis. Aliquam feugiat ultricies ipsum vel sollicitudin. Vivamus nulla massa, hendrerit sit amet nibh quis, porttitor convallis nisi. \\" />
      <Blueprint2.Button className=\\"assessment-briefing-button\\" onClick={[Function: onClick]} text=\\"Continue\\" />
    </Blueprint2.Card>
  </Blueprint2.Dialog>
  <Workspace controlBarProps={{...}} editorProps={{...}} editorWidth=\\"50%\\" handleEditorWidthChange={[Function: handleEditorWidthChange]} handleSideContentHeightChange={[Function: handleSideContentHeightChange]} hasUnsavedChanges={false} mcqProps={{...}} sideContentHeight={[undefined]} sideContentProps={{...}} replProps={{...}} />
</div>"
`;
